ocaml/xapi/xapi_xenops.ml

```diff
@@ -617,20 +617,21 @@ module MD = struct
     try
       let implementation =
         MxGPU {
+          physical_function = get_target_pci_address ~__context vgpu;
+          vgpus_per_pgpu =
+            List.assoc Xapi_globs.mxgpu_vgpus_per_pgpu internal_config
+            |> Int64.of_string;
           framebufferbytes;
-          sched =
-            List.assoc Xapi_globs.mxgpu_sched internal_config
-            |> int_of_string;
         }
       in {
         id = (vm.API.vM_uuid, vgpu.Db_actions.vGPU_device);
         position = int_of_string vgpu.Db_actions.vGPU_device;
         implementation;
       }
     with
-    | Not_found -> failwith "AMD MxGPU settings (sched slice) not specified"
-    | Failure "int_of_string" ->
-      failwith "AMD MxGPU settings (sched slice) invalid"
+    | Not_found -> failwith "AMD MxGPU settings not specified"
+    | _ ->
+      failwith "AMD MxGPU settings invalid"
```

Comment:
For debugging, I would include the string of the exception (`Printexc.to_string e`). I am surprised to see `failwith` here and not `error`, which is more convenient to use because it takes `printf`-style arguments.